### PR TITLE
[ticket/16675] Adjust checking commit messages

### PR DESCRIPTION
**Pull request by rxu**
Check commit messages on both 'push' and 'pull_request' events.

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16675">PHPBB3-16675</a>.

### CODE REVIEW

**Comment by CHItA** [comment]
Is there an example where the current version didn't work?

**Comment by rxu** [comment]
Sure, any merged pull requests in 3.3.x/master commits range on `push` event. F.e. https://github.com/phpbb/phpbb/runs/1664625037 where commit message check didn't run, and so on.

**Comment by CHItA** [comment]
I don't think we want to run the checks on push as those should only contain merge commits. One scenario where the changes in run might be useful though is on old PRs with updates without a rebase I think. As it is possible that the base commit is not in the 100 commits we check out.

**Comment by rxu** [comment]
I'm just following the current workflow where all other checks run on both `pull_request` and `push` events. Then probably duplicating any checks on `push` is redundant too as checks on `pull request` run on top of the base branch anyway.

**Comment by CHItA** [comment]
Pushes can have can have merge conflict resolutions etc. Thus our unit and functional tests are a bit different in that regard. So IMO we could keep the `if` block, but the other changes look good to me. I added Marc, so he can figure this out.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -77,12 +77,10 @@ jobs:
...
             - name: Check commit message
-              env:
-                  BRANCH: ${{ github.event.pull_request.base.sha }}
-                  PR_SHA: ${{ github.event.pull_request.head.sha }}
-              if: github.event.pull_request.head.sha != '' && github.event.pull_request.base.sha != ''
               run: |
-                  git-tools/commit-msg-hook-range.sh $BRANCH..$PR_SHA
+                  # Get previous merge commit to get commit range from
+                  export BRANCH=$(git rev-list --merges -n 1 HEAD^1)
+                  git-tools/commit-msg-hook-range.sh $BRANCH..$GITHUB_SHA
```
Comment (CHItA) [inline .github/workflows/tests.yml:83]: Where does `$GITHUB_SHA` comes from?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -77,12 +77,10 @@ jobs:
                   .github/check-executable-files.sh ./
 
             - name: Check commit message
-              env:
-                  BRANCH: ${{ github.event.pull_request.base.sha }}
-                  PR_SHA: ${{ github.event.pull_request.head.sha }}
-              if: github.event.pull_request.head.sha != '' && github.event.pull_request.base.sha != ''
```
Comment (CHItA) [inline .github/workflows/tests.yml:83]: I would go with `if: github.event.pull_request.head.sha != ''`

File: .github/workflows/tests.yml (hunk 1 above)
Comment (rxu) [reply .github/workflows/tests.yml:83]: It's default GA env var https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables

File: .github/workflows/tests.yml (hunk 2 above)
Comment (rxu) [reply .github/workflows/tests.yml:83]: Or probably just `if: env.GITHUB_EVENT_NAME == 'pull_request'`

Comment (CHItA) [reply .github/workflows/tests.yml:83]: That might be more explicit.

**Comment by rxu** [comment]
Changed the code, or do we want to keep `export BRANCH=$(git rev-list --merges -n 1 HEAD^1)` definition? I'm confused a bit.

**Comment by CHItA** [comment]
> Changed the code, or do we want to keep `export BRANCH=$(git rev-list --merges -n 1 HEAD^1)` definition? I'm confused a bit.

That change looks good to me.

**Review by marc1706 — APPROVED** [review]

**Pull request merged**